content/blog/2017/2017-02-14-declarative-notifications.adoc

```diff
@@ -0,0 +1,436 @@
+
+== Adding Notifications
+
+Next I'll pull in the code for the `sendNotification` method from the original blog post.
+I wrote the method to only depend on the one parameter and the Steps from the
+three notification plugins.
+Like I said, Declarative is still Pipeline -
+we'll just copy and paste this code in and,
+since I'm running on the same master with the same configuration,
+it should just work.
+
+[source, groovy]
+----
+pipeline {
+  /* ... unchanged ... */
+}
+
+def sendNotifications(String buildStatus = 'STARTED') {
+  // build status of null means successful
+  buildStatus =  buildStatus ?: 'SUCCESSFUL'
+
+  // Default values
+  def colorName = 'RED'
+  def colorCode = '#FF0000'
+  def subject = "${buildStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]'"
+  def summary = "${subject} (${env.BUILD_URL})"
+  def details = """<p>STARTED: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]':</p>
+    <p>Check console output at &QUOT;<a href='${env.BUILD_URL}'>${env.JOB_NAME} [${env.BUILD_NUMBER}]</a>&QUOT;</p>"""
+
+  // Override default values based on build status
+  if (buildStatus == 'STARTED') {
+    color = 'YELLOW'
+    colorCode = '#FFFF00'
+  } else if (buildStatus == 'SUCCESSFUL') {
+    color = 'GREEN'
+    colorCode = '#00FF00'
+  } else {
+    color = 'RED'
+    colorCode = '#FF0000'
+  }
+
+  // Send notifications
+  slackSend color: colorCode, message: summary
+
+  hipchatSend color: color, notify: true, message: summary
+
+  emailext (
+      to: 'bitwiseman@bitwiseman.com',
+      subject: subject,
+      body: details,
+      recipientProviders: [[$class: 'DevelopersRecipientProvider']]
+    )
+}
+----
+
+image::/images/post-images/2017-02-14/blueocean-notifications.png[Global Pipeline Library, role="center"]
+
+== Move Notifications to Shared Library
+
+Our current pipeline is pretty good, we have our Declarative Pipeline sending notifications.
+However, the notification method taking up about a third of our `Jenkinsfile` is a bit of a distraction.
+Also, as Jenkins admin, I'd rather share that notification method among all my projects
+without copying it to each new project and then having maintain all the copies.
+I want to move that method to a Shared Library.
+
+Shared Libraries are not specific to Declarative. They were released in their
+current form several months ago and were functional and useful in Scripted Pipeline.
+Not every method like this one should go into a Shared Library,
+but Declarative's focus on overall pipeline flow makes it much clearer what
+parts of a pipeline are candidates for moving to a shared library.
+
+
+=== Setup a Shared Library
+
+I've created a simple shared library repository for this series of posts, called
+link:https://github.com/bitwiseman/jenkins-pipeline-shared[bitwiseman/jenkins-pipeline-shared].
+The Shared Library functionality has too many configuration options to cover in one post.
+I've chosen to set mine up a "Global" Library, accessible from any project on my Jenkin master.
```

Comment:
Grammar weidsies and typos, rephrase to "I've chosen to mine up as a "Global Pipeline Library," accessible from any project on my Jenkins master.""

Generally speaking, please try to use the exact text from the web UI when referring to specific entities in blog posts and documentation.